File: week07/18/prob5.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

bool searchMatrix(int** matrix, int matrixSize, int* matrixColSize, int target){
    int m = matrixSize;
    int n = *matrixColSize;

    for (int i=0;i<m;i++){
        for (int j=0;j<n;j++){
            if (matrix[i][j] == target){
                return true;}
            else if (matrix[i][j] > target){
                break;}
        }
    }

    return false;
}

File: week07/23/prob5.py
class Solution:
    def searchMatrix(self, mat, target: int):
        
        m=len(mat)
        n=len(mat[0])
        
        i=m-1
        j=0
        
        while i>=0 and j<n:
            if mat[i][j]==target:
                return True
            elif mat[i][j]<target:
                j+=1
            else:
                i-=1
                
        return False